custom_metrics/privacy.js

```diff
@@ -1,24 +1,141 @@
+      tcData = null;
+    }
+    return tcData;
+  })(),
+  iab_usp: (() => {
+    // IAB US Privacy User Signal Mechanism “USP API” is integrated on the page.
+    // https://github.com/InteractiveAdvertisingBureau/USPrivacy
+    // Test site: nfl.com
+    let uspData;
+    if (typeof __uspapi == 'function') {
+      // based on https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#gettcdata
+      __uspapi('getUSPData', 1, (result, success) => {
+        if(success) {
+          uspData = result;
+        } else {
+          uspData = {"error": "Failed to retrieve USPData"};
+        }
+      });
+    } else {
+      uspData = null;
+    }
+    return uspData;
+  })(),
+
+  // Ads Transparency Spotlight Data Disclosure schema
+  // Check `meta` tag cf. https://github.com/Ads-Transparency-Spotlight/documentation/blob/main/implement.md
+  // Only for top frame; can't access child frames (same-origin policy)
+  // Test site: ?
+  ads_transparency_spotlight: (() => {
+    let ats_meta_tag = document.querySelector('meta[name="AdsMetadata"]');
+    if (ats_meta_tag) {
+      return ats_meta_tag.content; // JSON message with metadata
+    } else {
+      return null;
+    }
+  })(),
+
+  // FLoC origin trial
+  // Check `meta` tag cf. https://developer.chrome.com/blog/floc/ (complement with relevant header later on)
+  // Test site: https://floc-ot-meta.glitch.me/
+  floc_origin_trial: (() => {
+    let floc_meta_tag = document.querySelector('meta[http-equiv="origin-trial"]');
+    if (floc_meta_tag) {
+      return floc_meta_tag.content; // Get origin trial token, could be useful to check if multiple enrolled sites are from the same webmaster
+    } else {
+      return null;
+    }
+  })(),
+
+  // Check function/variable accesses through string searches (wrappers cannot be used, as the metrics are only collected at the end of the test)
+  // FLoc - test site: https://floc.glitch.me/
+  floc_document_interestCohort: findStringInBodies('document.interestCohort'),
+  floc_interestCohort: findStringInBodies('interestCohort'),
+  // DNT - test site: https://www.theverge.com/
+  navigator_doNotTrack: findStringInBodies('navigator.doNotTrack'),
+  doNotTrack: findStringInBodies('doNotTrack'),
+  // GPC - test site: https://global-privacy-control.glitch.me/
+  navigator_globalPrivacyControl: findStringInBodies('navigator.globalPrivacyControl'),
+  globalPrivacyControl: findStringInBodies('globalPrivacyControl'),
+  // sensitive resources - accesses permissions policy
+  document_permissionsPolicy: findStringInBodies('document.permissionsPolicy'),
+  permissionsPolicy: findStringInBodies('permissionsPolicy'),
+  // user media (camera, microphone)
+  navigator_mediaDevices_getUserMedia: findStringInBodies('navigator.mediaDevices.getUserMedia'),
+  mediaDevices: findStringInBodies('mediaDevices'),
+  getUserMedia: findStringInBodies('getUserMedia'),
+  // geolocation
```

Comment:
@VictorLeP Why did you decide to separate different matches?

Reply:
(I'm assuming you're referring to `mediaDevices` and `getUserMedia`)
In general, I match different (sub)strings to balance false negative and false positive matches.
That said, the match on `mediaDevices` is probably not very useful, as `getUserMedia` would capture all (relevant) accesses anyway. I'll leave it out.

Reply:
Updated in `a17248f`

Reply:
Maybe you could share how do you plan to query it for the chapter?
I had an idea to put it into some hierarchical form.
GEO API example:
```
geolocation: {
  web_api: {
    navigator_geolocation_getCurrentPosition: findStringInBodies('navigator.geolocation.getCurrentPosition') || findStringInBodies('navigator') && findStringInBodies('geolocation') && findStringInBodies('getCurrentPosition'),
    navigator_geolocation_watchPosition: findStringInBodies('navigator.geolocation.watchPosition') || findStringInBodies('navigator') && findStringInBodies('geolocation') && findStringInBodies('watchPosition'),
  }
}
```
And then we have just to work with booleans in SQL, light and easy.
What do you think?

Reply:
Yeah, sounds good - my approach would complicate things too much, as there is no way to consolidate the numbers for the different but related strings.
What would the `web_api` level be useful for though? I don't see why more detail would be needed there.

Reply:
I'm collecting a list of IP geolocation API, like https://ipstack.com/ mentioned in the outline.
But now I rather think to do a PR to: https://github.com/AliasIO/wappalyzer, right?

Is there any other way to get geolocation features?

Reply:
My idea was to just search requests for relevant URLs, for example for IPStack: https://api.ipstack.com/.
The only place where that would fail is some sort of 'open-source' IP geolocation library that you could host yourself, but my intuition is that these will be primarily paid services that host their own APIs/libraries.

Wappalyzer could be an alternative approach, but it feels like more work for little gain.

Moreover, the two types of geolocation should be well separated: geolocation APIs are 'hidden' and only use your IP, while `navigator.geolocation` is 'visible' as permission is requested, but the resulting data is much more precise.

Reply:
Overall, I agree with refactoring string searches to the or/and forms you proposed, but I don't see a lot of added value in the hierarchy - it seems to me like just another indirection to traverse in the query.

Reply:
Yeah, I totally agree here. Will change variable structure for:
- geolocation,
- mediaDevices,
- globalPrivacyControl,
- permissionsPolicy
- and interestCohort
based on the example then.